app/src/main/cpp/skyline/common/math_util.h

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: 2013 Dolphin Emulator Project
+// SPDX-FileCopyrightText: 2014 Citra Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <algorithm>
+#include <cstdlib>
+#include <type_traits>
+
+namespace Common {
+
+constexpr float PI = 3.1415926535f;
+
+template <class T>
+struct Rectangle {
+    T left{};
+    T top{};
+    T right{};
+    T bottom{};
+
+    constexpr Rectangle() = default;
+
+    constexpr Rectangle(T width, T height) : right(width), bottom(height) {}
+
+    constexpr Rectangle(T left_, T top_, T right_, T bottom_)
+        : left(left_), top(top_), right(right_), bottom(bottom_) {}
+
+    [[nodiscard]] constexpr T Left() const {
```

Comment:
We generally don't use getters for public member variables, you can drop all these